### PR TITLE
[OPIK-299] New dataset item migration using new migration template

### PR DESCRIPTION
**Pull request by thiagohora**
## Details
Retrying dataset item migration using the new data migration template.

## Issues
OPIK-299

## Testing
- Scripts tested manually

## Documentation
- The new data migration is documented via changelog and readme files.

### CODE REVIEW

File: apps/opik-backend/data-migrations/1.0.3/scripts/db-app-analytics/000001_data_migration_dataset_items_data_field.sql
Comment (andrescrz) [inline apps/opik-backend/data-migrations/1.0.3/scripts/db-app-analytics/000001_data_migration_dataset_items_data_field.sql:1]: You likely want to break this file into multiple ones, per each migration step. It'd be easier and safer to run each of the steps.

**Review by andrescrz — APPROVED** [review]
Generally LGTM. I just advice to break the migration scripts into multiple ones.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# 2024 Changelog
+
+### Release 1.0.3, 2024-10-29 [#](apps/opik-backend/data-migrations/1.0.3/README.md)
+
+#### Backward Incompatible Change
+
+- The structure of dataset items has changed to include new dynamic fields. In order to make previously created dataset
```
Comment (Nimrod007) [inline CHANGELOG.md]: I would inform users that:
```
The structure of dataset items has changed to include new dynamic fields. dataset items logged before version 1.0.3 will still show but would not be searchable. 

If you would like to migrate previous dataset items to the new format please see below instructions: [dataset item migration](apps/opik-backend/data-migrations/1.0.3/README.md) for more information*.
```

**Review by Nimrod007 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (thiagohora) [reply CHANGELOG.md]: Done

**Review by Nimrod007 — APPROVED** [review]

**Pull request merged**